accepted/future-releases/nnbd/number-operation-typing.md

```diff
@@ -111,11 +140,13 @@ These extensions also carry over to the compound assignment operators and increm
 
 An `lhs += e` expression is roughly equivalent to `lhs = lhs + e` except that subexpressions of `lhs` are only evaluated once. The static typing rules that apply to `lhs + e` also applies to `lhs += e`, and similarly for the other binary operators that are special cased.
 
-That means that `int x = 0; x += 1;` works because `x += 2` is equivalent to `x =  x + 2`, and `x + 2` has type `int`.
+That means that `int x = 0; x += 2;` works because `x += 2` is equivalent to `x =  x + 2`, and `x + 2` has type `int`.
```

Comment:
Maybe cover the case `double d = 0; d += 2;` as well. `d += 2.0` will work fine, but with `2` we'd get context type `num` and `2` will have static type `int`, and `d = d + 2` has no errors.

But will it ever be a problem that `2` is an `int` here, when it will often have type `double` as soon as it gets within a few inches of anything double?

Reply:
Probably not a problem in practice.

The only situation I can see is:

```dart
double x = something;
x += 147573952589676410000;  // 2^67
```
This will fail to compile because the context is `num`. It could compile if we made the context `double`.
It would only make a difference for literals that are actually valid double values, and an optimizing compiler can do that without any rules.

So, we could make the context type of `e2` in `dbl + e2` be `double` because we know that the first operation on the value will be `toDouble()`. 

It would affect an integer literal: `dbl + 2` would become `dbl + 2.0`, but that would happen anyway.
And it would affect `dbl + await Future.value(2);` which would ... possibly ... make that `await Future.value(2.0)`, but again no real value.

I don't think it would change anything in practice, so let's not go there.

Reply:
Makes sense.